src/index.js

```diff
@@ -123,13 +123,17 @@ async function bundleFunctions(file, utils) {
     },
   };
 
-  const bundle = await rollup.rollup(options);
-  const {
-    output: [{ code }],
-  } = await bundle.generate({
-    format: "iife",
-  });
-  return code;
+  try {
+    const bundle = await rollup.rollup(options);
+    const {
+      output: [{ code }],
+    } = await bundle.generate({
+      format: "iife",
+    });
+    return code;
+  } catch (error) {
+    return utils.build.failBuild("Error while bundling Edge handlers", { error });
```

Comment:
will this still throw? what will the return value of the function become? don't we have to cancel upstream steps somehow?

Reply:
From a plugin authoring standpoint, the execution will stop right there.
The way this is implemented under the hood is by throwing an error.
Rollup should stop on failure, and uploading has not started yet. The build will stop, which should prevent anything else to be run. Is there something specific you had in mind that should be cancelled?

Reply:
i just wanted to make sure we don't continue uploading empty code when returning here

Reply:
can you add a comment that calling this actually throws and thereby stops execution?

Reply:
Just did :+1: